compiler/next/lib/uast/VarDecl.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/uast/VarDecl.h"
+
+#include "chpl/uast/Builder.h"
+
+namespace chpl {
+namespace uast {
+
+
+VarDecl::~VarDecl() {
+}
```

Comment:
Another general style related question is which member functions are we defining in headers, and which are we defining in cpp files?

This file felt a little too awkward as is :)

Reply:
@mppf to the rescue

Reply:
This one must be defined in the .cpp file AFAIK because it is a pure virtual function in the header but one that C++ requires us to define.

Reply:
Or maybe this is a @leekillough question? :D

Reply:
baseAST.h has `virtual ~BaseAST() = default;`
uAST's Decl.h has `virtual ~Decl() = 0;`

I think we should use one of these two forms in the .h file here, instead of `VarDecl::~VarDecl() { }`

Reply:
We can't, because C++ requires us to define it.

Reply:
E.g. see https://stackoverflow.com/questions/1219607/why-do-we-need-a-pure-virtual-destructor-in-c

> Note: The destructor is the only method that even if it is pure virtual has to have an implementation in order to instantiate derived classes (yes pure virtual functions can have implementations).

I do not claim to be an expert on these C++ things; it is merely my understanding that we are required to have it.

Reply:
A difference between the two - `virtual ~Decl() = 0;` is making the class pure virtual while `virtual ~BaseAST() = default;` would allow it to be constructed.